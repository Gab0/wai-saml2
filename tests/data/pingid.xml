<samlp:Response Version="2.0" ID="AovTvbYKoXbh0r24xsyBNCHcJER" IssueInstant="2023-12-22T18:14:18.809Z" Destination="http://host.minikube.internal:3000/sso/assert" xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol"><saml:Issuer xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion">authentication</saml:Issuer><ds:Signature xmlns:ds="http://www.w3.org/2000/09/xmldsig#"><ds:SignedInfo><ds:CanonicalizationMethod Algorithm="http://www.w3.org/2001/10/xml-exc-c14n#"/><ds:SignatureMethod Algorithm="http://www.w3.org/2001/04/xmldsig-more#rsa-sha256"/><ds:Reference URI="#AovTvbYKoXbh0r24xsyBNCHcJER"><ds:Transforms><ds:Transform Algorithm="http://www.w3.org/2000/09/xmldsig#enveloped-signature"/><ds:Transform Algorithm="http://www.w3.org/2001/10/xml-exc-c14n#"/></ds:Transforms><ds:DigestMethod Algorithm="http://www.w3.org/2001/04/xmlenc#sha256"/><ds:DigestValue>AFTJKElrLMvwRdjXnznwPXwt18+cXRu6MTu5HeD8eC8=</ds:DigestValue></ds:Reference></ds:SignedInfo><ds:SignatureValue>YqnsjSe1OKOJlL85CUwlNHaYnxl2INYBsCk2eBDT5+Km7AP+qZr4WNNEhjqgkBXEckN3I0dGKgWZ0VAa4/MVqixy1U/TXqseOpIdBzgmHrQD/w1uRrQDIdBysZMU9br7PSC/9ILPNSMcppTcU7wfntDOXNK3AxDBaKDzEZpPhOiRkqo5T3qv4OS7Q+43COlEz8A/bLwGUBg/E+6wQbo8vKzZWR4aRZr3IwNrOxbZyS5xyM19nhxThZteplAJslkDxySpx9pyD+KH7Re2vUKeFHTvWYG/I+ZPU+jFRyn1siwgWp2Av4bcC+aAy/G+OCBwwVZjcftKzW2kzKV8UxMdYwh6ZM0RewJHjD8H+EWzzpN3qx3/z81yYPg1giJphLd5jvJl+Mh+6WLL10sCJxA2JXLlaJdDGFXtgAL84LZ8Kq++4YMiKRAqE3Rpjn6Ys/jesVnH5m0K5P0BmsBkxbyP7qZk8q2d9e2bruQPHFLSSlvOYcSHYsg1mJJKxMim7q9gE2EduF4xo3+QnpuRRqZlNOTWV2sLqROkMJFdzikqcYVk22lco/VuSu6FB1A9f407F7eWpEa7jbL9dt7ASWN/fkLRktiVYG2InKok8vMNTdvF+U5vEotw4Dewfm48sfQcjzFNkyTOraEvXNS42AD5Oi4U13E6Uw6HtWg3fZOWGus=</ds:SignatureValue></ds:Signature><samlp:Status><samlp:StatusCode Value="urn:oasis:names:tc:SAML:2.0:status:Success"/></samlp:Status><saml:Assertion ID="SDcihfummt.7x.64J1owM_q9EuG" IssueInstant="2023-12-22T18:14:18.818Z" Version="2.0" xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion"><saml:Issuer>authentication</saml:Issuer><saml:Subject><saml:NameID Format="urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified">tester</saml:NameID><saml:SubjectConfirmation Method="urn:oasis:names:tc:SAML:2.0:cm:bearer"><saml:SubjectConfirmationData Recipient="http://host.minikube.internal:3000/sso/assert" NotOnOrAfter="2023-12-22T18:19:18.819Z"/></saml:SubjectConfirmation></saml:Subject><saml:Conditions NotBefore="2023-12-22T18:09:18.819Z" NotOnOrAfter="2023-12-22T18:19:18.819Z"><saml:AudienceRestriction><saml:Audience>postgrest</saml:Audience></saml:AudienceRestriction></saml:Conditions><saml:AuthnStatement SessionIndex="SDcihfummt.7x.64J1owM_q9EuG" AuthnInstant="2023-12-22T18:14:18.814Z"><saml:AuthnContext><saml:AuthnContextClassRef>urn:oasis:names:tc:SAML:2.0:ac:classes:unspecified</saml:AuthnContextClassRef></saml:AuthnContext></saml:AuthnStatement></saml:Assertion></samlp:Response>
