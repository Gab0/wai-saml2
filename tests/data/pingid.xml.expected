Response
  { responseDestination =
      "http://host.minikube.internal:3000/sso/assert"
  , responseInResponseTo = Nothing
  , responseId = "AovTvbYKoXbh0r24xsyBNCHcJER"
  , responseIssueInstant = 2023-12-22 18:14:18.809 UTC
  , responseVersion = "2.0"
  , responseIssuer = "authentication"
  , responseStatusCode =
      MkStatusCode
        { statusCodeValue = Success , statusCodeSubordinate = Nothing }
  , responseSignature =
      Signature
        { signatureInfo =
            SignedInfo
              { signedInfoCanonicalisationMethod = C14N_EXC_1_0
              , signedInfoSignatureMethod = RSA_SHA256
              , signedInfoReference =
                  Reference
                    { referenceURI = "AovTvbYKoXbh0r24xsyBNCHcJER"
                    , referenceDigestMethod = DigestSHA256
                    , referenceDigestValue =
                        "AFTJKElrLMvwRdjXnznwPXwt18+cXRu6MTu5HeD8eC8="
                    }
              }
        , signatureValue =
            "YqnsjSe1OKOJlL85CUwlNHaYnxl2INYBsCk2eBDT5+Km7AP+qZr4WNNEhjqgkBXEckN3I0dGKgWZ0VAa4/MVqixy1U/TXqseOpIdBzgmHrQD/w1uRrQDIdBysZMU9br7PSC/9ILPNSMcppTcU7wfntDOXNK3AxDBaKDzEZpPhOiRkqo5T3qv4OS7Q+43COlEz8A/bLwGUBg/E+6wQbo8vKzZWR4aRZr3IwNrOxbZyS5xyM19nhxThZteplAJslkDxySpx9pyD+KH7Re2vUKeFHTvWYG/I+ZPU+jFRyn1siwgWp2Av4bcC+aAy/G+OCBwwVZjcftKzW2kzKV8UxMdYwh6ZM0RewJHjD8H+EWzzpN3qx3/z81yYPg1giJphLd5jvJl+Mh+6WLL10sCJxA2JXLlaJdDGFXtgAL84LZ8Kq++4YMiKRAqE3Rpjn6Ys/jesVnH5m0K5P0BmsBkxbyP7qZk8q2d9e2bruQPHFLSSlvOYcSHYsg1mJJKxMim7q9gE2EduF4xo3+QnpuRRqZlNOTWV2sLqROkMJFdzikqcYVk22lco/VuSu6FB1A9f407F7eWpEa7jbL9dt7ASWN/fkLRktiVYG2InKok8vMNTdvF+U5vEotw4Dewfm48sfQcjzFNkyTOraEvXNS42AD5Oi4U13E6Uw6HtWg3fZOWGus="
        , signatureKeyInfo = Nothing
        }
  , responseAssertion =
      Just
        Assertion
          { assertionId = "SDcihfummt.7x.64J1owM_q9EuG"
          , assertionIssued = 2023-12-22 18:14:18.818 UTC
          , assertionIssuer = "authentication"
          , assertionSubject =
              Subject
                { subjectConfirmations =
                    [ SubjectConfirmation
                        { subjectConfirmationMethod = Bearer
                        , subjectConfirmationAddress = ""
                        , subjectConfirmationNotOnOrAfter = 2023-12-22 18:19:18.819 UTC
                        , subjectConfirmationRecipient =
                            "http://host.minikube.internal:3000/sso/assert"
                        }
                    ]
                , subjectNameID =
                    NameID
                      { nameIDQualifier = Nothing
                      , nameIDSPNameQualifier = Nothing
                      , nameIDSPProvidedID = Nothing
                      , nameIDFormat = Just Unspecified
                      , nameIDValue = "tester"
                      }
                }
          , assertionConditions =
              Conditions
                { conditionsNotBefore = 2023-12-22 18:09:18.819 UTC
                , conditionsNotOnOrAfter = 2023-12-22 18:19:18.819 UTC
                , conditionsAudienceRestrictions =
                    [ AudienceRestriction
                        { audienceRestrictionAudience = [ "postgrest" ] }
                    ]
                }
          , assertionAuthnStatement =
              AuthnStatement
                { authnStatementInstant = 2023-12-22 18:14:18.814 UTC
                , authnStatementSessionIndex = "SDcihfummt.7x.64J1owM_q9EuG"
                , authnStatementLocality = ""
                }
          , assertionAttributeStatement = []
          }
  , responseEncryptedAssertion = Nothing
  }